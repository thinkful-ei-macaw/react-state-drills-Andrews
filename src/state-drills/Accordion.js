import React, { Component } from 'react'

class Accordion extends Component {

    static defaultProps = {
        sections: []
    }
    
    state = {
        currentTabIndex: null
    };
    
    handleButtonClick(index) {
        this.setState({ currentTabIndex: index })
    };

    render() {
        return (
            <ul>
                {this.props.sections.map( (section, index) => ( 
                <li key={index}>
                    <button
                    onClick={() => this.handleButtonClick(index)}
                    >
                    {section.title}
                    </button>
                    {index == this.state.currentTabIndex ? (<p>{section.content}</p>) : "" }
                    
                </li>))
                }
            </ul>
        )
    }
}

export default Accordion;