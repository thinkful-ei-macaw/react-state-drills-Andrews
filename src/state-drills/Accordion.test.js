import React from 'react'
import Accordion from './Accordion'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

it('renders empty when sections prop not supplied', () => {
    expect(
        renderer.create(<Accordion />).toJSON()
        ).toMatchSnapshot()
  })

it('renders no sections as active by default', () => {
expect(
    renderer.create(<Accordion sections={sections}/>).toJSON()
    ).toMatchSnapshot()
})

it('opens a clicked section', () => {
        const wrapper = shallow(<Accordion sections={sections}/>)
        wrapper.find('button').at(1).simulate('click')
        expect(toJson(wrapper)).toMatchSnapshot()
    })

it('only opens the last section when multiple sections have been clicked', () => {
    const wrapper = shallow(<Accordion sections={sections}/>)
    wrapper.find('button').at(0).simulate('click')
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
})