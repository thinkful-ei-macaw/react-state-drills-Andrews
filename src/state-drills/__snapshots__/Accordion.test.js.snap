// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`only opens the last section when multiple sections have been clicked 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    <p>
      Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
    </p>
  </li>
</ul>
`;

exports[`opens a clicked section 1`] = `
<ul>
  <li
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`renders empty when sections prop not supplied 1`] = `<ul />`;

exports[`renders no sections as active by default 1`] = `
<ul>
  <li>
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    
  </li>
  <li>
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
    
  </li>
</ul>
`;
